Build fails with binary dependency but without attoparsec

In `src/Text/Parser/Input.hs`:
The imports of `Lazy` and `ByteString` are guarded by an `#ifdef MIN_VERSION_attoparsec`. Later on (from line 460 on), a part of the code guarded by `#ifdef MIN_VERSION_binary` also makes use of these imports.

Example errors:
```
src/Text/Parser/Input.hs:462:41: error:
    Not in scope: type constructor or class ‘Lazy.ByteString’
    Neither ‘Control.Monad.Trans.RWS.Lazy’,
            ‘Control.Monad.Trans.State.Lazy’ nor ‘Control.Monad.Trans.Writer.Lazy’ exports ‘ByteString’.
    |
462 |    type ParserInput (Lazy Binary.Get) = Lazy.ByteString
    |                                         ^^^^^^^^^^^^^^^

src/Text/Parser/Input.hs:470:43: error:
    Not in scope: type constructor or class ‘ByteString’
    |
470 |    type ParserInput (Strict Binary.Get) = ByteString
    |                                           ^^^^^^^^^^
```

--- a/src/Text/Parser/Input.hs
+++ b/src/Text/Parser/Input.hs
@@ -25,6 +25,9 @@ import qualified Control.Monad.Trans.State.Lazy as Lazy (StateT(StateT))
 import qualified Control.Monad.Trans.State.Strict as Strict (StateT(StateT))
 import qualified Control.Monad.Trans.RWS.Lazy as Lazy (RWST(RWST))
 import qualified Control.Monad.Trans.RWS.Strict as Strict (RWST(RWST))
+import Data.ByteString (ByteString)
+import qualified Data.ByteString as ByteString
+import qualified Data.ByteString.Lazy as Lazy
 import Data.Functor ((<$>))
 import qualified Data.List as List
 import Data.Ord (Down)
@@ -47,11 +50,8 @@ import Data.Monoid.Textual (TextualMonoid)
 import Data.Semigroup.Cancellative (LeftReductive)
 
 #ifdef MIN_VERSION_attoparsec
-import Data.ByteString (ByteString)
 import Data.Text (Text)
-import qualified Data.ByteString as ByteString
 import qualified Data.ByteString.Char8 as ByteString.Char8
-import qualified Data.ByteString.Lazy as Lazy
 import qualified Data.Text as Text
 
 import qualified Data.Attoparsec.ByteString as Attoparsec

